tools/rum-models-generator/Sources/RUMModelsGeneratorCore/Input/ObjcInterop/ObjcInteropTypeReader.swift

```diff
@@ -0,0 +1,197 @@
+                        owner: objcClass,
+                        swiftProperty: swiftProperty
+                    )
+                    propertyWrapper.objcInteropType = try objcInteropType(for: swiftArray)
+                    return propertyWrapper
+                case let swifTypeReference as SwiftTypeReference:
+                    let referencedType = try resolve(swiftTypeReference: swifTypeReference)
+
+                    switch referencedType {
+                    case let swiftStruct as SwiftStruct:
+                        let propertyWrapper = ObjcInteropPropertyWrapperAccessingNestedStruct(
+                            owner: objcClass,
+                            swiftProperty: swiftProperty
+                        )
+                        propertyWrapper.objcNestedClass = ObjcInteropReferencedTransitiveClass(
+                            owner: propertyWrapper,
+                            managedSwiftStruct: swiftStruct
+                        )
+                        return propertyWrapper
+                    case let swiftEnum as SwiftEnum:
+                        let propertyWrapper = ObjcInteropPropertyWrapperAccessingNestedEnum(
+                            owner: objcClass,
+                            swiftProperty: swiftProperty
+                        )
+                        propertyWrapper.objcNestedEnum = ObjcInteropReferencedEnum(
+                            owner: propertyWrapper,
+                            managedSwiftEnum: swiftEnum
+                        )
+                        return propertyWrapper
+                    default:
+                        throw Exception.illegal("Illegal reference type: \(swifTypeReference)")
+                    }
+                default:
+                    throw Exception.unimplemented(
+                        "Cannot generate @objc property wrapper for: \(type(of: swiftProperty.type))"
+                    )
+                }
+            }
+
+        try objcClass.objcPropertyWrappers
+            .compactMap { $0 as? ObjcInteropPropertyWrapperForTransitiveType }
+            .forEach { propertyWrapper in
+                // Store `ObjcInteropTypes` created for property wrappers
+                outputObjcInteropTypes.append(propertyWrapper.objcTransitiveType)
+                if let transitiveClass = propertyWrapper.objcTransitiveType as? ObjcInteropClass {
+                    // Recursively generate property wrappers for each transitive `ObjcInteropClass`
+                    try generateTransitiveObjcInteropTypes(in: transitiveClass)
+                }
+            }
+    }
+
+    private func objcInteropType(for swiftType: SwiftType) throws -> ObjcInteropType {
+        switch swiftType {
+        case _ as SwiftPrimitive<Bool>,
+             _ as SwiftPrimitive<Double>,
+             _ as SwiftPrimitive<Int>,
+             _ as SwiftPrimitive<Int64>:
```

Comment:
Naive question, what about `UInt`/`UInt64` and even `UInt*`/`Int*`?

Reply:
I wanted to start small and focus only on the constructs we need for RUM models - to have defined and limited scope for this PR. I heavily use `throw Exception.unimplemented(...)` in all the cases that are currently not handled. If the next version of `rum-events-format` will require `UInt` support, we will receive a clear feedback from this tool that it's not supported. How does it sound 🙂?

Reply:
I wasn't sure about the types we already exposed so that answers the question.
In general I think that it's a pragmatic approach and that sounds good to me :)